File: httpserver/src/main/java/com/shengshijie/httpserver/RawResponse.java

package com.shengshijie.httpserver;

import com.google.gson.GsonBuilder;

public final class RawResponse<T> {

    private int code;
    private String message;
    private T data;


    public RawResponse(int code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    @Override
    public String toString() {
        return "Response{" +
                "code=" + code +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }

    public String toJSONString(){
        return new GsonBuilder().create().toJson(this);
    }

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }

    public static <T> RawResponse<T> ok(T data) {
        return new RawResponse<>(200, "ok", data);
    }

    public static RawResponse<Void> ok() {
        return new RawResponse(200, "ok", null);
    }

    public static <T> RawResponse<T> ok(String msg, T data) {
        return new RawResponse<>(200, msg, data);
    }

    public static RawResponse fail(String msg) {
        return new RawResponse<String>(500, msg, null);
    }
}

File: server/src/main/java/com/shengshijie/server/http/RawResponse.kt
package com.shengshijie.server.http

import com.shengshijie.server.ServerManager

internal data class RawResponse<T>(private val code: Int, private val message: String, private val data: T) {

    companion object {

        @JvmStatic
        fun <T> ok(data: T,msg: String): RawResponse<T> {
            return RawResponse(ServerManager.mServerConfig.successCode, msg, data)
        }

        @JvmStatic
        fun fail(code: Int, msg: String): RawResponse<Any> {
            return RawResponse(code, msg, Unit)
        }
    }

}